### ISSUE TITLE
Input scheduling capabilities 

### ISSUE THREAD

**Issue by Jeffail** [Jeffail/benthos#580]
For some use cases it might be necessary to only consume from an input on a strict schedule, effectively making them batched inputs. This could mean periodically consuming from a directory, or perhaps reading a topic from start to finish. It would be great if we could solve this generally across all inputs by allowing them to be wrapped similarly to `read_until`.

Some changes will be need to be made as to how inputs trigger their consumer routines as this will need to be delayed until the schedule triggers. We should also factor in the possibility of coordinating instances of Benthos across a shared schedule, ideally via our existing cache interface for maximum flexibility.

This is tagged v4 as I'd like to at least know what the plan is before we kick off the next major version bump, in case there are fundamentals that need adjusting.

This issue is related to but not blocked by nor blocking: https://github.com/Jeffail/benthos/issues/505

**Comment by disintegrator** [comment]
#1069 might overlap in terms of high level design of how to control reading from inputs.

**Comment by DoisKoh** [comment]
Yes please. My use case: I'm trying to use Benthos to write to `csv` files and then upload them to S3 daily.

**Comment by twmb** [comment]
Input scheduling is now largely achievable via the `generate` input with `cron` interval, combined with processors that trigger the actual data fetching. Closing as the core use case is addressed. If you need more granular scheduling capabilities, feel free to open a new issue or PR.
